src/librustdoc/html/render/span_map.rs

```diff
@@ -73,6 +74,22 @@ impl<'tcx> SpanMapVisitor<'tcx> {
                 Some(def_id)
             }
             Res::Local(_) => None,
+            Res::PrimTy(p) => {
+                let span = path_span.unwrap_or_else(|| path.span);
+                if let Some(prim) = self
+                    .tcx
+                    .sess
+                    .source_map()
+                    .span_to_snippet(span)
```

Comment:
Please do not use span_to_snippet, it's not reliable (e.g. it will break for libstd if people use --remap-path-prefix). Why not use PrimitiveType::from unconditionally?

Reply:
I don't remember the reason so let's just use it unconditionnally.